--- a/src/btop_shared.hpp
+++ b/src/btop_shared.hpp
@@ -42,5 +42,6 @@
 #include <unistd.h>

+#include <pthread.h>
 // From `man 3 getifaddrs`: <net/if.h> must be included before <ifaddrs.h>
 // clang-format off
 #include <net/if.h>
@@ -269,7 +270,12 @@
 	struct ifaddrs* ifaddr;
 	int status;
 public:
-	IfAddrsPtr() { status = getifaddrs(&ifaddr); }
+	IfAddrsPtr() {
+		int cs;
+		pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
+		status = getifaddrs(&ifaddr);
+		pthread_setcancelstate(cs, nullptr);
+	}
 	~IfAddrsPtr() noexcept { freeifaddrs(ifaddr); }
 	IfAddrsPtr(const IfAddrsPtr &) = delete;
 	IfAddrsPtr& operator=(IfAddrsPtr& other) = delete;
